### DISCUSSION TITLE
A way to define the entire app in JSON or ???

### DISCUSSION THREAD

**Discussion by jfmatth** [dokku/dokku#4981] [Q&A]
Hi,

First - THANK YOU for making Dokku, this is such a cool app and so easy to use - THANK YOU.

I'm trying to keep my entire app in code, including all the dokku config.  Since I'm not a HEROKU user, I don't' know all the 'tricks' that dokku uses of Heroku's files.

is there a good reference in the docs that point to this?  They seem to be here and there, but didn't find a good list.

For example, the .env file, learned about that due to the NGINX buildpack.  is the .ENV a buildpack thing?  not sure, don't know.

**Comment by josegonzalez** [comment]
Unfortunately, not all config can be defined within the repository for various reasons - as an example, storing environment variable values in a repository in plaintext is bad practice, so we don't support it.

You might wish to look into [ansible-dokku](https://github.com/dokku/ansible-dokku), which provides a way to define your app in a single file.

**Reply by daitangio** [reply]
So there is no way to put some infrastructure information like [docker container options](https://dokku.com/docs/advanced-usage/docker-options/) inside some sort of configuration file?
Must we run the dokku command on target host every time?

Just asking because I was unable to find other ways in documentation.

And as far as I can understand, docker-compose is not supported at all.

**Comment by jfmatth** [comment]
Thank you.
